### PR TITLE
refactor(ia): audience development setup and tabs

### PR DESCRIPTION
**Pull request by miguelpeixe**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

p1732114314318549-slack-C06MPMZMF29

Introduces tabbed navigation for the Audience Development wizard.

The "Advanced Settings" toggle has also been removed and the sections that weren't placed under a tab are always visible below the RAS wizard.

image

Tabs should only become visible after RAS activation.

image

image

image

image

### How to test the changes in this Pull Request:

1. Delete the `newspack_reader_activation_enabled` option (CLI: `wp option delete newspack_reader_activation_enabled`)
2. Make sure you have WooCommerce Memberships available to test the Content Gating tab
3. Visit "Audience -> Setup" and confirm the pending RAS wizard renders without any tabs
4. Complete the setup and confirm you are redirected back but now with all the tabs and additional sections below the RAS setup
5. With WooCommerce Memberships deactivated, confirm the "Content Gating" tab is not available
6. Activate the plugin and confirm the tab is now available
7. Visit "Content Gating" and confirm the "Display memberships on the subscriptions tab" toggle works and the selection persists (refresh the page)
8. Click to Configure your gate and confirm you are redirected to the gate editor
9. Back to the wizard, visit the "Transactional Emails" tab
10. Confirm it renders as the image above
11. Click to edit the emails and confirm you are redirected to the editor

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: src/wizards/audience/views/setup/campaign.js [hunk 1]
```diff
@@ -72,7 +72,7 @@ export default withWizardScreen( () => {
 		setSkipped( { ...skipped, status: 'pending' } );
 		try {
 			const request = await apiFetch( {
-				path: '/newspack/v1/wizard/newspack-audience-configuration/reader-activation/skip-campaign-setup',
+				path: '/newspack/v1/wizard/newspack-audience/reader-activation/skip-campaign-setup',
```
Comment (jaredrethman) [inline src/wizards/audience/views/setup/campaign.js]:
```suggestion
				path: '/newspack/v1/wizard/newspack-audience/reader-activation/skip-campaign',
```

**Review by jaredrethman — CHANGES_REQUESTED** [review]
@miguelpeixe thank you for the refactor, I think its much better. I made one suggestion below to fix RAS skipping endpoint.

One other nitpick - in `src/wizards/audience/views/setup/content-gating.js` inside a `useEffect` hook there's `window.scrollTo( 0, 0 );`, should we have these in other tab components for consistency (I've also seen the same `scrollTo` code in setup)? If you agree I've prepared a DRY [solution](https://github.com/Automattic/newspack-plugin/compare/epic/ia-audience-setup-tabs...feat/ia-audience-setup-tabs__scroll-to) for you to see/use.

File: src/wizards/audience/views/setup/campaign.js (hunk 1 above)
Comment (miguelpeixe) [reply src/wizards/audience/views/setup/campaign.js]: Thank you for catching that!

**Comment by miguelpeixe** [comment]
> One other nitpick - in src/wizards/audience/views/setup/content-gating.js inside a useEffect hook there's window.scrollTo( 0, 0 );, should we have these in other tab components for consistency (I've also seen the same scrollTo code in setup)? If you agree I've prepared a DRY [solution](https://github.com/Automattic/newspack-plugin/compare/epic/ia-audience-setup-tabs...feat/ia-audience-setup-tabs__scroll-to) for you to see/use.

The content gating inherited the `scrollTo()` from the `setup.js`. There's no need for it because there's no redirect back to that wizard, as there can be throughout the setup steps.

I think the wizard feature you propose might be overkill for now. I removed the call in 205d4fad912835588a19572a1f82fe9155f8e648

**Comment by jaredrethman** [comment]
> I think the wizard feature you propose might be overkill for now...

Got it. The same code can be seen in the Setup component. Although it might be there for a different reason!?
https://github.com/Automattic/newspack-plugin/blob/205d4fad912835588a19572a1f82fe9155f8e648/src/wizards/audience/views/setup/setup.js#L38-L42

**Review by jaredrethman — APPROVED** [review]

**Comment by miguelpeixe** [comment]
It exists in the `setup.js` because of the [handoff redirects](https://github.com/Automattic/newspack-plugin/blob/205d4fad912835588a19572a1f82fe9155f8e648/src/wizards/audience/components/prerequisite.tsx#L138-L167)/

**Comment by miguelpeixe** [comment]
Thanks for the review, @jaredrethman!

**Pull request merged**